server/tck/src/test/java/io/smallrye/graphql/test/apps/subscription/api/StockTickerPublisher.java

```diff
@@ -0,0 +1,120 @@
+package io.smallrye.graphql.test.apps.subscription.api;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.reactivestreams.Publisher;
+
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.observables.ConnectableObservable;
+
+public class StockTickerPublisher {
+
+    private final Flowable<Stock> publisher;
+
+    public StockTickerPublisher() {
+        Observable<Stock> stockPriceUpdateObservable = Observable.create(emitter -> {
+
+            ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
+            executorService.scheduleAtFixedRate(newStockTick(emitter), 0, 2, TimeUnit.SECONDS);
+
+        });
+
+        ConnectableObservable<Stock> connectableObservable = stockPriceUpdateObservable.share().publish();
+        connectableObservable.connect();
+
+        publisher = connectableObservable.toFlowable(BackpressureStrategy.BUFFER);
+    }
+
+    public Publisher<Stock> getPublisher() {
+        return publisher;
+    }
+
+    public Publisher<Stock> getPublisher(List<String> stockCodes) {
+        return publisher.filter(stock -> stockCodes.contains(stock.stockCode));
+    }
+
+    private Runnable newStockTick(ObservableEmitter<Stock> emitter) {
+        return () -> {
+            List<Stock> stockPriceUpdates = getUpdates(rollDice(0, 5));
+            if (stockPriceUpdates != null) {
```

Comment:
another minor thing: `stockPriceUpdates` can never be `null`

Reply:
Yea, I still want to add cases where it can, but will remove this